src/renderer/viz/expressions/buckets.js

```diff
@@ -90,41 +92,55 @@ export default class Buckets extends BaseExpression {
         let children = {
             input
         };
+
         list.elems.map((item, index) => children[`arg${index}`] = item);
         super(children);
         this.numCategories = list.elems.length + 1;
         this.list = list;
         this.type = 'category';
+        this.isBuckets = true;
     }
+
     eval(feature) {
         const v = this.input.eval(feature);
-        let i;
-        for (i = 0; i < this.list.elems.length; i++) {
-            if (this.input.type == 'category' && v == this.list.elems[i].eval(feature)) {
-                return i;
-            } else if (this.input.type == 'number' && v < this.list.elems[i].eval(feature)) {
-                return i;
+        let i = 0;
+
+        if (this.input.type === 'category') {
+            for (i = 0; i < this.list.elems.length; i++) {
+                if (v === this.list.elems[i].eval(feature)) {
+                    return i;
+                }
             }
         }
+
+        if (this.input.type === 'number') {
+            for (i = 0; i < this.list.elems.length; i++) {
+                if (v < this.list.elems[i].eval(feature)) {
+                    return i;
+                }
+            }
+        }
+
         return i;
     }
+
     _compile(metadata) {
         super._compile(metadata);
-        this.othersBucket = this.input.type == 'category';
```

Comment:
Yep, also unnecessary

Reply:
I don't understand your reply :confused:

Reply:
Ups, I read it wrong, I thought you were asking me this property to be removed. After reading `ramp.js`, do you still think we need it?

Reply:
Is not needed after the refactor in `ramp.js`, yet my point is that this is a regression from the quality of the code point of view. Instead of having a global property for every expression and being generic this PR has checks like `instanceof Top` and `instaceof Buckets`

Reply:
I agree. Since we are going to change that in a short term, I would refactor that in a different PR because this code will change.

Reply:
I'll buy it